packages/nextjs/app/page.tsx

```diff
@@ -1,10 +1,92 @@
 "use client";
 
+import { useEffect, useState } from "react";
 import Link from "next/link";
 import type { NextPage } from "next";
 import { BugAntIcon, MagnifyingGlassIcon } from "@heroicons/react/24/outline";
+import { useScaffoldReadContract } from "~~/hooks/scaffold-eth";
 
 const Home: NextPage = () => {
+  const [counter, setCounter] = useState<number | undefined>();
+
+  const {
+    data,
+    error,
+    isLoading: contractLoading,
+  } = useScaffoldReadContract({
+    contractName: "BatchRegistry",
+    functionName: "checkedInCounter",
+  });
+
+  useEffect(() => {
+    if (data !== undefined) {
+      setCounter(Number(data));
+    }
+  }, [data]);
+
+  const renderContent = () => {
+    if (error) {
+      return (
+        <div className="p-4 rounded-lg bg-red-100 text-red-700" role="alert">
+          <p>Error fetching contract data: {error.message}</p>
+        </div>
+      );
+    }
+
+    return (
+      <p className="text-lg flex gap-2 justify-center">
+        <span className="font-bold">Checked in builders count:</span>
+        <span>{counter !== undefined ? counter : "0"}</span>
+      </p>
+    );
+  };
+
+  if (contractLoading) {
+    return (
+      <div className="w-full h-full mt-20 animate-pulse">
+        <div className="mb-6 flex items-center">
+          <div className="w-10 h-10 bg-gray-200 rounded-full mr-4"></div>
+          <div>
+            <div className="h-4 bg-gray-200 rounded w-24 mb-2"></div>
+            <div className="h-6 bg-gray-200 rounded w-48"></div>
+          </div>
+        </div>
+
+        <div className="grid md:grid-cols-2 gap-8">
+          <div className="bg-gray-200 rounded-lg h-96"></div>
+
+          <div>
+            <div className="h-8 bg-gray-200 rounded w-3/4 mb-4"></div>
+            <div className="space-y-2 mb-6">
+              {[...Array(6)].map((_, index) => (
+                <div key={index} className="h-4 bg-gray-200 rounded w-full"></div>
+              ))}
+            </div>
+
+            <div className="flex items-center mb-4">
+              <div className="h-8 bg-gray-200 rounded w-24 mr-4"></div>
+              <div className="h-8 bg-gray-200 rounded-full w-8"></div>
+              <div className="h-6 bg-gray-200 rounded w-8 mx-4"></div>
+              <div className="h-8 bg-gray-200 rounded-full w-8"></div>
+            </div>
+
+            <div className="flex space-x-4 mb-6">
+              <div className="h-10 bg-gray-200 rounded w-2/3"></div>
+              <div className="h-10 bg-gray-200 rounded w-1/3"></div>
+            </div>
+
+            <div>
+              <div className="h-6 bg-gray-200 rounded w-24 mb-2"></div>
+              <div className="h-4 bg-gray-200 rounded w-full mb-2"></div>
+              <div className="h-4 bg-gray-200 rounded w-full mb-2"></div>
+              <div className="h-4 bg-gray-200 rounded w-3/4"></div>
+            </div>
+          </div>
+        </div>
+      </div>
+    );
+  }
```

Comment:
Instead of having this "skeleton" you might handle the loading inline in renderContent